requirements/customer/scheduling-algorithm.tex

```diff
@@ -0,0 +1,48 @@
+\subsection{Scheduling Algorithm}
+This section describes how the central server distributes jobs to work machines: the \textit{scheduling algorithm}.
+Put briefly, the scheduling algorithm takes a list of jobs and decides which ones to run on the available work machines.
+The scheduling algorithm has the following goals:
+\begin{itemize}
+\item Keep work machines busy: there should be no idle work machines if there are pending jobs that could be run on them.
+\item Consider job priorities: jobs with high priorities should be worked on first.
+\item Consider time of job submission: jobs that were submitted first should be worked on first (if priorities are equal).
+\end{itemize}
+The scheduling algorithm operates under the following conditions:
+\begin{itemize}
+\item Jobs arrive at unforeseen times.
+\item Time required to complete jobs is unknown.
+\item Jobs cannot be transferred from one work machine to another.
+\item Jobs cannot be canceled and restarted.
```

Comment:
I think jobs can be cancelled (for example by the user or admin) but not restarted.

Reply:
I think he means that the algorithm should not cancel or restart jobs.

Reply:
In any case this needs a bit more clarification.

Reply:
What I meant is that jobs cannot be restarted once they've been canceled.

Reply:
I understood that, but I think it should be clarified.

Reply:
oh nvm I was looking at older commits.